### PR TITLE
Refs #33318 - add a script to filter package.json

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: script/filter-package-json.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+def is_excluded(package):
+    return (package in excludes['EXCLUDE_NPM_PACKAGES']
+            or any(package.startswith(prefix) for prefix in excludes['EXCLUDE_NPM_PREFIXES']))
+
+def filter_section(packages, section):
+    deps = packages.get(section, {})
+    for package, version in sorted(deps.items()):
+        if not is_excluded(package):
+            yield (package, version)
```
Comment (ekohl) [inline script/filter-package-json.py]: Just a thought:
```suggestion
def filter_section(packages, section):
    deps = packages.get(section, {})
    return {package: version for package, version in sorted(deps.items()) if not is_excluded(package)}
```

You could also make deps a direct parameter and do `data[section]` a level higher. So:
```suggestion
def filter_dependencies(deps):
    return {package: version for package, version in sorted(deps.items()) if not is_excluded(package)}
```

**Review by ekohl — DISMISSED** [review]

File: script/filter-package-json.py (hunk 1 above)
Comment (evgeni) [reply script/filter-package-json.py]: I like the later!

File: script/filter-package-json.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    return (package in excludes['EXCLUDE_NPM_PACKAGES']
+            or any(package.startswith(prefix) for prefix in excludes['EXCLUDE_NPM_PREFIXES']))
+
+def filter_dependencies(deps):
+    return {package: version for package, version in sorted(deps.items()) if not is_excluded(package)}
+
+with open('package.json') as package_json:
+    data = json.load(package_json)
+for section in ('devDependencies', 'dependencies'):
+    data[section] = dict(filter_dependencies(data.get(section, {})))
```
Comment (ekohl) [inline script/filter-package-json.py]: This is now sufficient.
```suggestion
    data[section] = filter_dependencies(data.get(section, {}))
```

File: script/filter-package-json.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+# workaround for https://projects.theforeman.org/issues/33319
+data['scripts']['postinstall'] = data['scripts']['postinstall'].replace('./script/npm-fix-foreman-stories.sh && ', '')
```
Comment (ekohl) [inline script/filter-package-json.py]: https://github.com/theforeman/foreman/pull/8730 is merged so this can go.

**Review by ekohl — APPROVED** [review]

**Pull request merged**